fvm/src/call_manager/mod.rs

```diff
@@ -191,3 +199,47 @@ pub struct FinishRet {
     pub events: Vec<StampedEvent>,
     pub events_root: Option<Cid>,
 }
+
+#[derive(Clone, Debug, Copy)]
+pub enum Entrypoint {
+    Invoke(MethodNum),
+    Upgrade(UpgradeInfo),
+}
+
+impl std::fmt::Display for Entrypoint {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Entrypoint::Invoke(method) => write!(f, "invoke({})", method),
+            Entrypoint::Upgrade(_) => write!(f, "upgrade"),
+        }
+    }
+}
+
+impl Entrypoint {
+    fn method_num(&self) -> MethodNum {
+        match self {
+            Entrypoint::Invoke(num) => *num,
+            Entrypoint::Upgrade(_) => fvm_shared::METHOD_UPGRADE,
+        }
+    }
+
+    fn func_name(&self) -> &'static str {
+        match self {
+            Entrypoint::Invoke(_) => "invoke",
+            Entrypoint::Upgrade(_) => "upgrade",
+        }
+    }
+
+    fn into_params(self, br: &mut BlockRegistry) -> Result<Vec<wasmtime::Val>> {
+        match self {
+            Entrypoint::Invoke(_) => Ok(Vec::new()),
+            Entrypoint::Upgrade(ui) => {
+                let ui_params = to_vec(&ui).map_err(
+                    |e| crate::syscall_error!(IllegalArgument; "failed to serialize upgrade params: {}", e),
+                )?;
+                let block_id = br.put_reachable(kernel::Block::new(CBOR, ui_params, Vec::new()))?;
```

Comment:
Can we add a comment on why not `DAG_CBOR`? Specifically, the linked code isn't (currently) considered reachable.